kafka-connect-deadletter/src/main/java/io/tabular/iceberg/connect/deadletter/DefaultFailedRecordFactory.java

```diff
@@ -0,0 +1,152 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.deadletter;
+
+import java.util.Map;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+public class DefaultFailedRecordFactory implements FailedRecordFactory {
+  private static final String DEAD_LETTER_TABLE_NAME_PROP = "table_name";
+
+  private static final String DEAD_LETTER_ROUTE_FIELD_PROP = "route_field";
+  private static final ConfigDef CONFIG_DEF =
+      new ConfigDef()
+          .define(
+              DEAD_LETTER_TABLE_NAME_PROP,
+              ConfigDef.Type.STRING,
+              null,
+              ConfigDef.Importance.MEDIUM,
+              "dead letter table name namespace.table")
+              .define(DEAD_LETTER_ROUTE_FIELD_PROP,
+                      ConfigDef.Type.STRING,
+                      null,
+                      ConfigDef.Importance.MEDIUM,
+                      "route field to inject table name on");
+
+  private static final String HEADERS = "headers";
+  private Schema schema;
+
+  private String deadLetterTableName;
+  private String deadLetterRouteField;
+
+  @Override
+  public Schema schema(String context) {
+    return schema;
+  }
+
+  @Override
+  public SinkRecord recordFromSmt(SinkRecord original, Throwable error, String context) {
+    Struct struct = new Struct(schema);
+    addCommon(struct, original, error, context);
+
+    if (original.key() != null) {
+      struct.put("key_bytes", original.key());
+    }
+    if (original.value() != null) {
+      struct.put("value_bytes", original.value());
+    }
+    if (!original.headers().isEmpty()) {
+      struct.put(HEADERS, DeadLetterUtils.serializedHeaders(original));
+    }
+
+    return original.newRecord(
+        original.topic(),
+        original.kafkaPartition(),
+        null,
+        null,
+        schema,
+        struct,
+        original.timestamp());
+  }
+
+  @Override
+  public SinkRecord recordFromConnector(SinkRecord record, Throwable error, String context) {
+
+    Struct struct = new Struct(schema);
+    addCommon(struct, record, error, context);
```

Comment:
Instead of setting context via the method, this should be fine. 

```suggestion
  public SinkRecord recordFromConnector(SinkRecord record, Throwable error) {

    Struct struct = new Struct(schema);
    addCommon(struct, record, error, "CONNECTOR_FAILURE");
```

Reply:
Removed all this CONTEXT stuff.